linker/linker.cpp

```diff
@@ -327,6 +327,11 @@ static void soinfo_free(soinfo* si) {
   // clear links to/from si
   si->remove_all_links();
 
+  // release all segments in rand_addr_segments
+  for (soinfo::SegmentInfo &seg_info : si->rand_addr_segments) {
+    munmap(reinterpret_cast<void*>(seg_info.real_addr), seg_info.page_size);
```

Comment:
I've renamed the fields. Do they make more sense now? I think I'm too close to this to evaluate if they make sense from an external perspective.